Formula/n/netatalk.rb

```diff
@@ -0,0 +1,72 @@
+class Netatalk < Formula
+  desc "File server for Macs, compliant with Apple Filing Protocol (AFP)"
+  homepage "https://netatalk.io"
+  url "https://github.com/Netatalk/netatalk/releases/download/netatalk-4-0-8/netatalk-4.0.8.tar.xz"
+  sha256 "d09d591521b27b91b1c2a6255a2e059af8c9fda43570f983f0a145706e5f7628"
+  license all_of: [
+    "GPL-2.0-only",
+    "GPL-2.0-or-later",
+    "LGPL-2.0-only",
+    "LGPL-2.1-or-later",
+    "BSD-2-Clause",
+    "BSD-3-Clause",
+    "MIT",
+  ]
+  head "https://github.com/Netatalk/netatalk.git", branch: "main"
+
+  depends_on "docbook-xsl" => :build
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkgconf" => :build
+
+  depends_on "berkeley-db@5" # macOS bdb library lacks DBC type etc.
+  depends_on "cracklib"
+  depends_on "libevent"
+  depends_on "libgcrypt"
+  depends_on "mysql"
+  depends_on "openldap" # macOS LDAP.Framework is not fork safe
+
+  uses_from_macos "libxslt" => :build
+
+  uses_from_macos "krb5"
+  uses_from_macos "libxcrypt"
+  uses_from_macos "perl"
+
+  on_linux do
+    depends_on "avahi" # on macOS we use native mDNS instead
+    depends_on "cups" # used by the AppleTalk print server
+    depends_on "libtirpc" # on macOS we use native RPC instead
+  end
+
+  def install
+    bdb5_rpath = rpath(target: Formula["berkeley-db@5"].opt_lib)
+    ENV.append "LDFLAGS", "-Wl,-rpath,#{bdb5_rpath}" if OS.linux?
+    args = [
+      "-Dwith-afpstats=false",
+      "-Dwith-appletalk=#{OS.linux?}", # macOS doesn't have an AppleTalk stack
+      "-Dwith-bdb-path=#{Formula["berkeley-db@5"].opt_prefix}",
+      "-Dwith-docbook-path=#{Formula["docbook-xsl"].opt_prefix}/docbook-xsl",
+      "-Dwith-init-dir=#{prefix}",
+      "-Dwith-init-hooks=false",
+      "-Dwith-install-hooks=false",
+      "-Dwith-pam-config-path=#{etc}/pam.d",
+      "-Dwith-rpath=false",
+      "-Dwith-spotlight=false",
+    ]
+
+    system "meson", "setup", "build", *args, *std_meson_args
+    system "meson", "compile", "-C", "build", "--verbose"
+    system "meson", "install", "-C", "build"
+  end
+
+  service do
+    name macos: "io.netatalk.daemon", linux: "netatalk"
```

Comment:
What do these service files look like? In particular, where do they send logs, and what working directory do they use?

Reply:
- macOS: https://github.com/Netatalk/netatalk/blob/main/distrib/initscripts/macos.netatalk.plist.in
  - logs to `/tmp/netatalk.out` and `/tmp/netatalk.err`
- Linux: https://github.com/Netatalk/netatalk/blob/main/distrib/initscripts/systemd.netatalk.service.in

Reply:
We probably want to fix up some of those path references, since they'll likely be `Cellar` paths that aren't suitable for long-lived services

Reply:
@carlocab So you're arguing that the two /tmp paths for the macOS logs should be rewritten to start with `#{prefix}`, for instance `/opt/homebrew/Cellar/netatalk/4.0.8/var/log` ?

Or am I getting your comment backwards?

Reply:
On Linux, we log to journald by default, and not to a particular file.

The working directory is not controlled by the service files, but rather the build system. On aarch64 macOS, for instance, the working directory is `/opt/homebrew/Cellar/netatalk/4.0.8/var/netatalk`.

Reply:
The problem is the generated paths in the plist include the version number, e.g. `/opt/homebrew/Cellar/netatalk/4.0.8/bin/netatalkd`. The version-agnostic path to use would be `/opt/homebrew/opt/netatalk/bin/netatalkd`.

Reply:
Hm, never mind, I see now that there's a symlink in place that makes `/opt/homebrew/opt/netatalk/var` a stable path between installed versions.

Reply:
Actually strike that, I ran some empirical tests. With the current configuration, updating from one version to another will wipe the contents of `/opt/homebrew/opt/netatalk/var`, because it symlinks to f.e. `/opt/homebrew/Cellar/netatalk/4.0.8/var` which gets replaced in an upgrade.

What gets me the expected behavior is `-Dwith-statedir-path=#{var}` which stores the CNID database under `/opt/homebrew/var/netatalk/CNID`

Reply:
I'd still suggest using the stable `opt_bin` path instead of the versioned path in the service files. See the `inreplace` directive in the [`batt` formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/b/batt.rb) for an example. Something like: `inreplace "distrib/initscripts/macos.netatalk.plist.in", "@bindir@", opt_bin`

Reply:
I do like these paths better. Now I've pushed this, plus the same substitution for the sbin dirs in all init scripts.